javascript/ql/src/semmle/javascript/frameworks/Koa.qll

```diff
@@ -115,6 +115,26 @@ module Koa {
     override RouteHandler getRouteHandler() { result = ctx.getRouteHandler() }
   }
 
+  /**
+   * A Koa request source, accessed through the a request property of a
+   * generator route handler (deprecated in Koa 3).
+   */
+  private class GeneratorRequestSource extends HTTP::Servers::RequestSource {
+    RouteHandler rh;
+
+    GeneratorRequestSource() {
+      exists(DataFlow::FunctionNode fun | fun = rh |
+        fun.getFunction().isGenerator() and
```

Comment:
Does it have to be a generator function? Can't we just remove that requirement? 

Looking at the [koajs.com](https://koajs.com/) it seems common to use async functions. 

Otherwise, if the evaluation turns out good, then this LGTM.

Reply:
I played around with `async function`, `function`, and `function*` as I could not find the definitive documentation for this behaviour. Only the `function*` variant exposes this field (and a console warning about deprecation).